clio-server/src/test/scala/org/broadinstitute/clio/server/service/IndexServiceSpec.scala

```diff
@@ -0,0 +1,118 @@
+package org.broadinstitute.clio.server.service
+
+import akka.stream.scaladsl.Sink
+import io.circe.Json
+import org.broadinstitute.clio.server.dataaccess.elasticsearch.{
+  ClioDocument,
+  ElasticsearchIndex
+}
+import org.broadinstitute.clio.server.{MockClioApp, TestKitSuite}
+import org.broadinstitute.clio.server.dataaccess.{MemoryPersistenceDAO, MemorySearchDAO}
+import org.broadinstitute.clio.transfer.model.TransferIndex
+import org.broadinstitute.clio.util.model.{DocumentStatus, UpsertId}
+
+abstract class IndexServiceSpec[
+  TI <: TransferIndex, D <: ClioDocument: ElasticsearchIndex
+](specificService: String)
+    extends TestKitSuite(specificService + "Spec") {
+
+  val memoryPersistenceDAO = new MemoryPersistenceDAO()
+  val memorySearchDAO = new MemorySearchDAO()
+
+  val indexService: IndexService[TI, D] = {
+    val app = MockClioApp(
+      searchDAO = memorySearchDAO,
+      persistenceDAO = memoryPersistenceDAO
+    )
+    val searchService = SearchService(app)
+    val persistenceService = PersistenceService(app)
+    getService(persistenceService, searchService)
+  }
+
+  val index: ElasticsearchIndex[D]
+  val dummyKey: indexService.transferIndex.KeyType
+  val dummyInput: indexService.transferIndex.QueryInputType
+
+  def getDummyMetadata(
+    documentStatus: Option[DocumentStatus]
+  ): indexService.transferIndex.MetadataType
+
+  def getService(
+    persistenceService: PersistenceService,
+    searchService: SearchService
+  ): IndexService[TI, D]
+  def copyDocumentWithUpsertId(originalDocument: D, upsertId: UpsertId): D
+  def documentToJson(document: D): Json
```

Comment:
Do we need this? I think we should be able to replace it with `document.asJson(index.encoder)`.

Reply:
tried that initially, but can't access .asJson from a non-specific ClioDocument.

Reply:
`asJson` is pulled in from a circe import, it's not tied to the specific document types. Try with
`import io.circe.syntax._`